src/odemis/acq/align/tdct.py

```diff
@@ -0,0 +1,146 @@
+    fm_image = convert_das_to_numpy_stack(das)
+
+    try:
+        # getzGauss can fail, so we need to catch the exception
+        zval, z, _ = multi_channel_get_z_guass(image=fm_image, x=x, y=y, show=show)
+        logging.info(f"Using Z-Gauss optimisation: {z}, previous z: {prev_z}")
+
+    except RuntimeError as e:
+        logging.warning(f"Error in z-gauss optimisation: {e}, using previous z: {prev_z}")
+        z = prev_z
+        x, y = prev_x, prev_y
+
+    return z
+
+
+def run_tdct_correlation(fib_coords: numpy.ndarray,
+                           fm_coords: numpy.ndarray,
+                           poi_coords: numpy.ndarray,
+                           fib_image: model.DataArray,
+                           fm_image: model.DataArray,
+                           path: str) -> Dict[str, Tuple[float, float]]:
+    """Run 3DCT Multi-point correlation between FIB and FM images.
+    :param fib_coords: the FIB coordinates (x, y, z)
+    :param fm_coords: the FM coordinates (x, y, z)
+    :param poi_coords: the point of interest coordinates
+    :param fib_image: the FIB image
+    :param fm_image: the FM image
+    :param path: the path to save the results
+    :return: the correlation results (including poi, transformations, and errors)
+    """
+
+    # get rotation center
+    halfmax_dim = int(max(fm_image.shape) * 0.5)
+    rotation_center = (halfmax_dim, halfmax_dim, halfmax_dim)
+
+    # get fib pixel size (meters)
+    fib_pixel_size = fib_image.metadata[model.MD_PIXEL_SIZE][0]
+
+    # fib image shape minus metadata, fib_pixelsize (microns), fm_image_shape
+    image_props = [fib_image.shape, fib_pixel_size * 1e6, fm_image.shape]
+
+    assert fib_image.ndim == 2, "FIB Image must be 2D"
+    assert fm_image.ndim == 3, "FM Image must be 3D"
+    assert fib_pixel_size is not None, "FIB Pixel Size must be set"
+    assert rotation_center is not None, "Rotation Center must be set"
+    assert isinstance(rotation_center, tuple), "Rotation Center must be a tuple"
+    assert len(rotation_center) == 3, "Rotation Center must have 3 values"
+
+    logging.debug(
+        f"Running 3DCT correlation with FIB image shape: {fib_image.shape}, FM image shape: {fm_image.shape}"
+    )
+
+    # run correlation
+    correlation_results = run_correlation(
+        fib_coords=fib_coords,
+        fm_coords=fm_coords,
+        poi_coords=poi_coords,
+        image_props=image_props,
+        rotation_center=rotation_center,
+        path=path,
+        fib_image_filename="FIB-Image",
+        fm_image_filename="FM-Image",
+    )
+
+    return correlation_results
+
+def get_poi_coordinate(correlation_results: dict) -> Tuple[float, float]:
+    """Get the the point of interest coordinate from correlation data
+    and convert from micrometers to meters in the microscope image coordinate system.
+    :param correlation_results: the correlation results
+    :return: the point of interest coordinate in meters
```

Comment:
Trying to connect the dots, it seems that you can use this function on the result returned by `run_tdct_correlation()`. Is that correct? What's the origin of the coordinate system? top-left of the image?

Reply:
it's centred at the image centre